routes/service/[service]/component/[component]/[operationType]/op/[operationName]/delete.tsx

```diff
@@ -0,0 +1,77 @@
+import { Handlers } from "$fresh/src/server/types.ts";
+import {
+  getAttachedPipeline,
+  getAudienceFromParams,
+} from "../../../../../../../../lib/utils.ts";
+import {
+  deleteAudience,
+  detachPipeline,
+} from "../../../../../../../../lib/mutation.ts";
+import { ResponseCode } from "snitch-protos/protos/sp_common.ts";
+import { serviceSignal } from "../../../../../../../../components/serviceMap/serviceSignal.ts";
+
+export const handler: Handlers<> = {
+  async POST(req, ctx) {
+    const audience = getAudienceFromParams(ctx.params);
+    const pipelines = serviceSignal.value.pipelines;
+    const config = serviceSignal.value.config;
+    const attachedPipeline = await getAttachedPipeline(
+      audience,
+      pipelines,
+      config,
+    );
+    if (attachedPipeline) {
```

Comment:
I think these nested if else's make the code difficult to read...and they contain a lot of duplicate code. Since the response building logic is the same for each termination, I think you can flatten this out and just build the response at the end. 

maybe something like:

```
let response;
if (attachedpipeline) {
response =  detachPipeline()
}
if (response?.code !== ResponseCode.ERROR) {
response = deleteAudience()
}
```
return build response.

Reply:
Thanks! this was adjusted to match that pattern